include/audio/dmic.h

```diff
@@ -0,0 +1,301 @@
+ *
+ * Returns the map of PDM controller and LEFT/RIGHT channel shifted to
+ * the bit position corresponding to the input logical channel value
+ *
+ * @param channel The logical channel number
+ * @param pdm The PDM hardware controller number
+ * @param lr LEFT/RIGHT channel within the chosen PDM hardware controller
+ *
+ * @return Bit-map containing the PDM and L/R channel information
+ */
+static inline u32_t dmic_build_channel_map(u8_t channel, u8_t pdm,
+		enum pdm_lr lr)
+{
+	return ((((pdm & BIT_MASK(3)) << 1) | lr) <<
+			((channel & BIT_MASK(3)) * 4));
+}
+
+/**
+ * Helper function to parse the channel map in pdm_chan_cfg
+ *
+ * Returns the PDM controller and LEFT/RIGHT channel corresponding to
+ * the channel map and the logical channel provided as input
+ *
+ * @param channel_map_lo Lower order/significant bits of the channel map
+ * @param channel_map_hi Higher order/significant bits of the channel map
+ * @param channel The logical channel number
+ * @param pdm Pointer to the PDM hardware controller number
+ * @param lr Pointer to the LEFT/RIGHT channel within the PDM controller
+ *
+ * @return none
+ */
+static inline void dmic_parse_channel_map(u32_t channel_map_lo,
+		u32_t channel_map_hi, u8_t channel, u8_t *pdm, enum pdm_lr *lr)
+{
+	u32_t channel_map;
+
+	channel_map = (channel < 8) ? channel_map_lo : channel_map_hi;
+	channel_map >>= ((channel & BIT_MASK(3)) * 4);
+
+	*pdm = (channel >> 1) & BIT_MASK(3);
+	*lr = channel & BIT(0);
+}
+
+/**
+ * Build a bit map of clock skew values for each PDM channel
+ *
+ * Returns the bit-map of clock skew value shifted to the bit position
+ * corresponding to the input PDM controller value
+ *
+ * @param pdm The PDM hardware controller number
+ * @param skew The skew to apply for the clock output from the PDM controller
+ *
+ * @return Bit-map containing the clock skew information
+ */
+static inline u32_t dmic_build_clk_skew_map(u8_t pdm, u8_t skew)
+{
+	return ((skew & BIT_MASK(4)) << ((pdm & BIT_MASK(3)) * 4));
+}
+
+/**
+ * Configure the DMIC driver and controller(s)
+ *
+ * Configures the DMIC driver device according to the number of channels,
+ * channel mapping, PDM I/O configuration, PCM stream configuration, etc.
+ *
+ * @param dev Pointer to the device structure for DMIC driver instance
+ * @param cfg Pointer to the structure containing the DMIC configuration
+ *
+ * @return 0 on success, a negative error code on failure
+ */
+static inline int dmic_configure(struct device *dev, struct dmic_cfg *cfg)
+{
+	const struct _dmic_ops *api = dev->driver_api;
+
+	return api->configure(dev, cfg);
+}
+
+/**
+ * Send a command to the DMIC driver
+ *
+ * Sends a command to the driver to perform a specific action
+ *
+ * @param dev Pointer to the device structure for DMIC driver instance
+ * @param cmd The command to be sent to the driver instance
+ *
+ * @return 0 on success, a negative error code on failure
+ */
+static inline int dmic_trigger(struct device *dev, enum dmic_trigger cmd)
```

Comment:
IMHO dmic_cmd() is a better name. 
Also, there is no callback informing when given command is executed (I assume that some controllers might delay stop command until a memory buffer will be fully filled).

Reply:
This follows the convention used in existing I2S driver, DMA driver and potentially other drivers. I prefer to be consistent with other drivers.